ruby/ql/lib/codeql/ruby/experimental/Rbi.qll

```diff
@@ -150,22 +150,14 @@ module Rbi {
     class ProcCall extends RbiType, SignatureCall, MethodCallAgainstT {
       ProcCall() { this.getMethodName() = "proc" }
 
-      private ProcReturnsTypeCall getReturnsTypeCall() { result.getProcCall() = this }
-
-      private ProcParamsCall getParamsCall() { result.getProcCall() = this }
+      override ReturnsTypeCall getReturnsTypeCall() {
+        result.(ProcReturnsTypeCall).getProcCall() = this
+      }
 
-      /**
-       * Gets the return type of this type signature.
-       */
-      override ReturnType getReturnType() { result = this.getReturnsTypeCall().getReturnType() }
+      override ProcParamsCall getParamsCall() { result.getProcCall() = this }
 
-      /**
-       * Gets the type of a parameter of this type signature.
-       */
-      override ParameterType getAParameterType() {
-        result = this.getParamsCall().getAParameterType()
-      }
-      // TODO: get associated method to which this can be passed
+      // TODO: widen type for procs/blocks
+      override MethodBase getAssociatedMethod() { none() }
```

Comment:
Should this return `Callable` for procs/blocks/lambdas?

Reply:
We'd probably want this to return a `BlockParameter`, e.g. in:
```
  sig do
    params(
      a: String,
      block: T.nilable(T.proc.params(b: Integer).void)
    ).void
  end
  def blk_nilable_method(a, &block); end
```
the `T.proc` call corresponds to the `&block` parameter of `blk_nilable_method`, whereas the `sig` call corresponds to the `blk_nilable_method` definition itself. So we maybe need something like `newtype TTypedCallable = TypedMethod(MethodBase m) or TypedBlock(BlockParameter b)` or maybe to add another predicate `BlockParameter getAssociatedBlockParameter()` to the `SignatureCall` abstract class and in practice only one of `getAssociatedMethod()` and `getAssociatedBlockParameter()` will have a value.